### ISSUE TITLE
Memory OOB access in try-catch

### ISSUE THREAD

**Issue by xermicus** [hyperledger/solang#1292]
The following contract traps the execution with a `MemoryOutOfBounds` trap code:

```solidity
contract c {
    constructor() payable {}

    function test() public returns (int32) {
        int32 x = 0;
        try new other(true) {
            x = 1;
        } catch (bytes memory _c) {
            x = 2;
        }
        return x;
    }
}

contract other {
    constructor(bool rev) public {
        if (rev) {
            revert("foo");
        }
    }

    function _ext() public {}
}
```

A real substrate node will not tell about the OOB error, however the mock runtime will panic on this. See substrate test `calls::try_catch_constructor`. Uncommenting the last function call to the `test` function in this test will trigger the OOB access. 

 `cargo test --test substrate -- calls::try_catch_constructor`

**Comment by anj20** [comment]
@xermicus can I work on this?

**Comment by xermicus** [comment]
Sure, all yours

**Comment by anj20** [comment]
here we can use a dynamic byte memory '_c' outside the catch function this will store the error message when the catch function is triggered. the test function code will be:-
```rust
function test() public returns (int32) {
    int32 x = 0;
    bytes memory _c;
    try new other(true) {
        x = 1;
    } catch (bytes memory c) {
        _c = c;
        x = 2;
    }
    return x;
}

```

**Comment by anj20** [comment]
@xermicus ?

**Comment by xermicus** [comment]
How's that related to the problem?

**Comment by xermicus** [comment]
Closed by #1415
